Contract changes

karsrhyder opened the pull request:
- [x] When the state changes: Fire an event 
```
ItemChange(bytes32 indexed itemHash, status newstatus, address providerAddress)
```

- [x] Change event to
```
event NewItem(address owner, bytes32 itemHash, string ipfsMetadata, uint itemValue, uint hashtagFee, uint seekerRep);
```

- [x] Split readItem into
  - [x] readItemState: ```status, providerAddress, providerRep, numberOfReplies```
  - [x] readItemData: ```itemValue, seekerAddr, seekerRep, ipfsMetadata, creationBlock```

- [x] On item creation, store the ```creationBlock``` in the struct, so it can be returned in the ```readItemData``` call.

karsrhyder committed:
HashtagList test added

The new hashtag is added to the hashtagList contract. The test reads out the hashtagList.

karsrhyder committed:
Working!

karsrhyder committed:
Added HashtagList integration

When a new hashtag is created, it will add itself to the hashtagList contract provided in the constructor params.

karsrhyder committed:
Contracts

I'm going to try to verify this on etherscan

karsrhyder committed:
Asymmetrical Simple Deal Hashtag Contract

Commit:
Safety commit

Commit:
Replies on the Hashtagcontract

Commit:
RepliesOnHashtag

Commit:
Merge remote-tracking branch 'origin/cleanerContract' into cleanerContract

karsrhyder committed:
Contract changes

- [x] When the state changes: Fire an event 
```
ItemChange(bytes32 indexed itemHash, status newstatus, address providerAddress)
```

- [x] Change event to
```
event NewItem(address owner, bytes32 itemHash, string ipfsMetadata, uint itemValue, uint hashtagFee, uint seekerRep);
```

- [x] Split readItem into
  - [x] readItemState: ```status, providerAddress, providerRep, numberOfReplies```
  - [x] readItemData: ```itemValue, seekerAddr, seekerRep, ipfsMetadata, creationBlock```

- [x] On item creation, store the ```creationBlock``` in the struct, so it can be returned in the ```readItemData``` call.

dapplion reviewed (approved):
Looks good! After testing the overlapping features we will decide which ones to keep

dapplion committed:
Re-structured data

- metadataHashes are stored as bytes32 instead of string
- each reply's metadataHash are stored in an array

dapplion committed:
Small fixes to simpleDeal

karsrhyder committed:
readRepliers

dapplion committed:
make it ERC677 compatible

kingflurkel committed:
Merge pull request #19 from swarmcity/erc677_compatible

make it ERC677 compatible

karsrhyder committed:
Tests for bridged token version of simpleDeal
